tests/rptest/tests/admin_uuid_operations_test.py

```diff
@@ -82,18 +95,292 @@ def _uuids_updated():
                 old_uuid = n['uuid']
 
         # get current node id and UUID
-        current = admin.get_broker_uuid(to_stop)
+        current = self.admin.get_broker_uuid(to_stop)
 
-        admin.override_node_id(to_stop,
-                               current_uuid=current['node_uuid'],
-                               new_node_id=initial_to_stop_id,
-                               new_node_uuid=old_uuid)
+        self.admin.override_node_id(to_stop,
+                                    current_uuid=current['node_uuid'],
+                                    new_node_id=initial_to_stop_id,
+                                    new_node_uuid=old_uuid)
 
         self.redpanda.restart_nodes(to_stop,
                                     auto_assign_node_id=True,
                                     omit_seeds_on_idx_one=False)
 
-        after_restart = admin.get_broker_uuid(to_stop)
+        after_restart = self.admin.get_broker_uuid(to_stop)
 
         assert after_restart['node_id'] == initial_to_stop_id
         assert after_restart['node_uuid'] == old_uuid
+
+    def scrape_uuid(self, node: ClusterNode) -> str | None:
+        UUID_LOG = "'Generated new UUID for node'"
+        lines = [
+            s.strip() for s in self.log_searcher._capture_log(node, UUID_LOG)
+        ]
+        if len(lines) < 1:
+            return None
+        self.logger.info(f"UUID Lines: {json.dumps(lines, indent=1)}")
+        assert len(lines) == 1, f"Too many: {json.dumps(lines, indent=1)}"
+        return lines[0].split(":")[-1].strip()
+
+    def _restart_node(self,
+                      node: ClusterNode,
+                      overrides: dict | None = None,
+                      extra_cli: list[str] = [],
+                      drop_disk: bool = False):
+        self.redpanda.stop_node(node)
+        if drop_disk:
+            self.redpanda.clean_node(node,
+                                     preserve_current_install=True,
+                                     preserve_logs=False)
+
+        self.redpanda.start_node(
+            node,
+            auto_assign_node_id=True,
+            omit_seeds_on_idx_one=False,
+            override_cfg_params=overrides,
+            extra_cli=extra_cli,
+        )
+
+    def _decommission(self, node_id, node=None):
+        def decommissioned():
+            try:
+
+                results = []
+                for n in self.redpanda.nodes:
+                    if self.redpanda.node_id(n) == node_id:
+                        continue
+
+                    brokers = self.admin.get_brokers(node=n)
+                    for b in brokers:
+                        if b['node_id'] == node_id:
+                            results.append(b['membership_status'] != 'active')
```

Comment:
nit: it is better to wait until the node fully disappears ("draining" membership status means that the decom just started).